src/main/java/com/aws/greengrass/deployment/KernelUpdateDeploymentTask.java

```diff
@@ -72,6 +73,9 @@ public DeploymentResult call() {
             } else if (KERNEL_ROLLBACK.equals(stage)) {
                 result = new DeploymentResult(DeploymentResult.DeploymentStatus.FAILED_ROLLBACK_COMPLETE,
                         getDeploymentStatusDetails());
+            } else if (ROLLBACK_BOOTSTRAP.equals(stage)) {
+                result = new DeploymentResult(DeploymentResult.DeploymentStatus.FAILED_UNABLE_TO_ROLLBACK,
```

Comment:
Why is the status `FAILED_UNABLE_TO_ROLLBACK` in this case?

Reply:
When a bootstrap-on-rollback step fails, we do the following (from Kernel.java::241-253):

1. set the deployment object's error information
2. set the deployment object's DeploymentStage enum value to ROLLBACK_BOOTSTRAP
3. offer the deployment object into the DeploymentQueue
4. launch the kernel in DEFAULT stage

Thus, the conditional logic you see here (`if (ROLLBACK_BOOTSTRAP.equals(stage)) { result = new DeploymentResult(FAILED_UNABLE_TO_ROLLBACK); ...`.

If, on the other hand, bootstrap-on-rollback had succeeded, the kernel would have progressed into the rollback activation stage (Kernel.java::257-271) and done the following:

1. set the deployment object's DeploymentStage enum value to KERNEL_ROLLBACK
2. offer the deployment object into the DeploymentQueue
3. launch the kernel in DEFAULT stage

Hence the preceding conditional logic (`if (KERNEL_ROLLBACK.equals(stage)) { result = new DeploymentResult(FAILED_ROLLBACK_COMPLETE); ...`.